Services/Authentication/classes/class.ilSession.php

```diff
@@ -252,7 +252,18 @@ public static function _destroy($a_session_id, $a_closing_context = null, $a_exp
         ilSessionIStorage::destroySession($a_session_id);
 
         $ilDB->manipulate($q);
-        
+
+        try {
+            // only delete session cookie if it is set in the current request
+            if (isset($_COOKIE[session_name()]) && $_COOKIE[session_name()] == $a_session_id) {
```

Comment:
Could we use `$DIC->http()->cookieJar()` instead (and for ILIAS >= 8.x `$DIC->http()->wrapper()->cookie()`, a PR for 8.x would be highly appreciated)?

Reply:
See comment above